server/src/main/java/com/cloud/projects/ProjectManagerImpl.java

```diff
@@ -656,7 +656,14 @@ public Project updateProject(final long projectId, final String displayText, fin
         Transaction.execute(new TransactionCallbackWithExceptionNoReturn<ResourceAllocationException>() {
             @Override
             public void doInTransactionWithoutResult(TransactionStatus status) throws ResourceAllocationException {
-                if (displayText != null) {
+                if (displayText != null && name != null) {
```

Comment:
in line 666, add condition `else if (displayText != null) {`

this part  can be optimized

Reply:
please consider what if name is null then still displayText can be null or not, both are no dealt with in the else clause.

Reply:
condition added and if name is null, displaytext can also be null.both are dealt now